bentoml/deployment/sagemaker/__init__.py

```diff
@@ -223,6 +225,44 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 }
 
 
+def parse_aws_client_exception_or_raise(e):
+    if type(e) == ClientError:
+        error_response = e.response.get('Error', {})
+        error_code = error_response.get('Code')
+        if error_code == 'ValidationException':
+            return Status.NOT_FOUND(error_response.get('Message', 'Unknown'))
+        elif error_code == 'InvalidSignatureException':
+            return Status.UNAUTHENTICATED(error_response.get('Message', 'Unknown'))
+        else:
+            raise e
+    elif type(e) == ParamValidationError:
+        return Status.INVALID_ARGUMENT(str(e))
+
+
+def _cleanup(client, name, version, cleanup_endpoint_config=False):
```

Comment:
separate this to `_cleanup_sagemaker_endpoint_config` and  `_cleanup_sagemaker_model`